### ISSUE TITLE
Add use of pendulum for date/datetime management

### ISSUE THREAD

**Issue by dreinon** [koxudaxi/datamodel-code-generator#643]
**Is your feature request related to a problem? Please describe.**
I recently found [pendulum](https://pendulum.eustace.io/) and started using it as my default date and datetime management library since the standard python datetime module is not very rich.

**Describe the solution you'd like**
It would be nice if we could choose either to use the standard datetime module or to use [pendulum](https://pendulum.eustace.io/) as our default library to type date/datetime properties.

I'm already using pendulum types in pydantic models I coded manually and so far so good! These types are `pendulum.Date` and `pendulum.DateTime` and `pendulum.Time`.

**Comment by koxudaxi** [comment]
@dreinon 
I'm sorry for my late reply.
It's interesting. I will add the task to a TODO list.
Thank you!!

**Comment by karolzlot** [comment]
Outdated statements are present on pendulum github page: https://github.com/sdispater/pendulum/pull/556

It may be better idea to support [Arrow](https://github.com/arrow-py/arrow), as it is currently maintained more than pendulum.

**Comment by dreinon** [comment]
@karolzlot I agrer with supporting arrow instead, but pendulum worked directly with pydantic. Do you think arrow wouldn't?

**Comment by karolzlot** [comment]
@dreinon
It works ok, but not as well as built-in datatime.
Example issue I had: https://stackoverflow.com/questions/68930298/how-to-use-arrow-type-in-fastapi-response-schema (proposed workaround works, but it's a hack and it removes Arrow from pydantic model)

**Comment by dreinon** [comment]
@karolzlot
Right, but looking at the pydantic codebase, I think this is more of a job Arrow needs to do instead of pydantic itself.

The following two lines explain that they only define [validators](https://github.com/samuelcolvin/pydantic/blob/f0acf6efe7001ffc8c9b8d8a7229153ae17f151f/pydantic/validators.py#L609) and [schema](https://github.com/samuelcolvin/pydantic/blob/63337fbadc2eccfd2942f21593cffc2adf92b084/pydantic/schema.py#L769) for standard library types. That's why pendulum worked out of the blue, since pendulum is a superset of the date-related standard library types.

In order to make Arrow pydantic-compatible, the following page from the pydantic docs can be followed, that says that any class can be made pydantic compatible by setting a couple [methods for validation and schema generation](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)

Hope this is helpful! Thank you 🙂

**Comment by karolzlot** [comment]
> In order to make Arrow pydantic-compatible, the following page from the pydantic docs can be followed, that says that any class can be made pydantic compatible by setting a couple [methods for validation and schema generation](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)

@dreinon I think it won't be enough. What is needed is in this PR: https://github.com/samuelcolvin/pydantic/issues/951 

Pydantic just doesn't have all needed features yet, although various workarounds are possible.

**Comment by vlcinsky** [comment]
Personally I have tried both Arrow (even with [small contribution](https://github.com/arrow-py/arrow/pull/118)) and pendulum.

My personal preference is pendulum and it works great to me for years.

Today I have found pendulum even supports time intervals/periods.

**Comment by Woody1193** [comment]
Not sure if anyone is aware of this, but [pydantic-extra-types](https://github.com/pydantic/pydantic-extra-types) exists so Pydantic already "supports" pendulum.

**Comment by ilovelinux** [comment]
@koxudaxi @gaborbernat 
It looks like the requested feature has been satisfied thanks to #1914 🙂
